modules/network/main.tf

```diff
@@ -3,8 +3,9 @@ data "aws_availability_zones" "available" {
 }
 
 locals {
-  public_subnets          = var.additional_subnet && !var.subnet_autocreate_as_private ? concat(var.public_subnets_cidr, [var.alb_additional_subnet_cidr_block]) : var.public_subnets_cidr
-  private_subnets         = var.additional_subnet && var.subnet_autocreate_as_private ? concat(var.private_subnets_cidr, [var.alb_additional_subnet_cidr_block]) : var.private_subnets_cidr
+  additional_subnet       = var.additional_subnet ? var.alb_additional_subnet_cidr_block : ""
+  subnets_cidrs           = concat(var.subnets_cidrs, [local.additional_subnet])
```

Comment:
This will not work right. If `additional_subnet = false` you will get list with empty string, something like ["1.1.1.1/1", ""]
`local.additional_subnet` isn't used, so no need for this local vae, you can just have:
  subnets_cidrs           = var.additional_subnet ? concat(var.subnets_cidrs, [var.alb_additional_subnet_cidr_block]) : var.subnets_cidrs